language/move-core/types/src/u256.rs

```diff
@@ -0,0 +1,270 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use ethnum::U256 as EthnumU256;
+use num::{bigint::Sign, BigInt};
+#[cfg(any(test, feature = "fuzzing"))]
+use proptest::{prelude::*, strategy::BoxedStrategy};
+use std::{
+    fmt,
+    ops::{BitAnd, BitOr, BitXor, Shl, Shr},
+};
+use uint::FromStrRadixErr;
+
+const U256_NUM_BYTES: usize = 32;
+pub type U256FromStrError = FromStrRadixErr;
+
+// This U256 impl was chosen for now but we are open to changing it as needed
+use primitive_types::U256 as PrimitiveU256;
+use serde::{ser::SerializeTuple, Deserialize, Deserializer, Serialize, Serializer};
+
+#[allow(non_camel_case_types)]
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Copy, PartialOrd, Ord)]
+pub struct U256Inner(PrimitiveU256);
```

Comment:
The name `U256Inner` was chosen on a whim. I'm open to other names. Maybe something like `U256Move`? Since we'll have to expose this type to users.

Reply:
We could have this be `U256` and just have the Move VM type be `U256(move_core_types::U256)`.